sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/DicomIO.java

```diff
@@ -0,0 +1,195 @@
+   * https://cloud.google.com/healthcare/docs/how-tos/pubsub. The connector will output a {@link
+   * ReadStudyMetadata.Result} which will contain metadata of the study encoded as a json array.
+   */
+  public static class ReadStudyMetadata
+      extends PTransform<PCollection<String>, ReadStudyMetadata.Result> {
+
+    private ReadStudyMetadata() {}
+
+    /** TupleTag for the main output. */
+    public static final TupleTag<String> METADATA = new TupleTag<String>() {};
+    /** TupleTag for any error response. */
+    public static final TupleTag<String> ERROR_MESSAGE = new TupleTag<String>() {};
+
+    public static class Result implements POutput, PInput {
+      private PCollection<String> readResponse;
+
+      private PCollection<String> failedReads;
+
+      /** Contains both the response and error outputs from the transformation. */
+      PCollectionTuple pct;
+
+      /**
+       * Create DicomIO.ReadStudyMetadata.Result from PCollectionTuple which contains the response
+       * (with METADATA and ERROR_MESSAGE tags).
+       *
+       * @param pct the pct
+       * @return the read result
+       * @throws IllegalArgumentException the illegal argument exception
+       */
+      static ReadStudyMetadata.Result of(PCollectionTuple pct) throws IllegalArgumentException {
+        if (pct.getAll()
+            .keySet()
+            .containsAll((Collection<?>) TupleTagList.of(METADATA).and(ERROR_MESSAGE))) {
+          return new ReadStudyMetadata.Result(pct);
+        } else {
+          throw new IllegalArgumentException(
+              "The PCollection tuple must have the DicomIO.ReadStudyMetadata.METADATA "
+                  + "and DicomIO.ReadStudyMetadata.ERROR_MESSAGE tuple tags");
+        }
+      }
+
+      private Result(PCollectionTuple pct) {
+        this.pct = pct;
+        this.readResponse = pct.get(METADATA);
+        this.failedReads = pct.get(ERROR_MESSAGE);
+      }
+
+      /**
+       * Gets failed reads.
+       *
+       * @return the failed reads
+       */
+      public PCollection<String> getFailedReads() {
+        return failedReads;
+      }
+
+      /**
+       * Gets resources.
+       *
+       * @return the resources
+       */
+      public PCollection<String> getReadResponse() {
+        return readResponse;
+      }
+
+      @Override
+      public Pipeline getPipeline() {
+        return this.pct.getPipeline();
+      }
+
+      @Override
+      public Map<TupleTag<?>, PValue> expand() {
+        return ImmutableMap.of(METADATA, readResponse);
+      }
+
+      @Override
+      public void finishSpecifyingOutput(
+          String transformName, PInput input, PTransform<?, ?> transform) {}
+    }
+
+    /**
+     * DoFn to fetch the metadata of a study from a Dicom store based on it's location and study id.
+     */
+    @SuppressWarnings({"nullness", "rawtypes"})
+    static class FetchStudyMetadataFn extends DoFn<String, String> {
+
+      private HealthcareApiClient dicomStore;
```

Comment:
I suspect that if you do this, you can remove the `nullness` condition in SuppressWarnings?

```suggestion
      private HealthcareApiClient dicomStore = new HttpHealthcareApiClient();
```

Reply:
Maybe you also need to add null checks in process... up to you.

Reply:
Hmm I'm still failing compileJava.